HesoyamHospital/GraphicEditor/SearchService.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Microsoft.EntityFrameworkCore.Internal;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Windows;
+
+namespace GraphicEditor
+{
+    class SearchService
+    {
+        public SearchService()
+        {
+        }
+
+        public List<MapLocation> Find_objects_by_name(string name) 
+        {
+            List<MapLocation> map_locations = new List<MapLocation>();
+            List<MapLocation> map_location1 = Find_in_file("hospital1groundfloor.txt", name);
+            List<MapLocation> map_location2 = Find_in_file("hospital1firstfloor.txt", name);
+            List<MapLocation> map_location3 = Find_in_file("hospital2groundfloor.txt", name);
+            List<MapLocation> map_location4 = Find_in_file("hospital2firstfloor.txt", name);
+            List<MapLocation> map_location5 = Find_in_file("hospital2secondfloor.txt", name);
+
+            map_locations.AddRange(map_location1);
+            
+            map_locations.AddRange(map_location2);
+
+            map_locations.AddRange(map_location3);
+            
+            map_locations.AddRange(map_location4);
+            
+            map_locations.AddRange(map_location5);
+
+            return map_locations;
+        }
+
+        public List<MapLocation> Find_in_file(string path, string name)
+        {
+            List<String> names = new List<string>();
+            MapLocation map_location = null ;
+            List<MapLocation> map_locations = new List<MapLocation>();
+            GraphicRepository graphic_repository = new GraphicRepository();
+            List<GraphicalObject> graphical_objects = graphic_repository.ReadFromFile(path);
+
+            foreach (GraphicalObject obj in graphical_objects) {
+                if (obj.Name.Contains(name)) 
+                {
+                    string full_name = obj.Name;
+                    map_location = Get_Location_By_Filename(path, full_name);
+                    map_locations.Add(map_location);
+                    
+                }
+            }
+
+            return map_locations;
+        
+        }
+
+        public MapLocation Get_Location_By_Filename(string path, string name) {
```

Comment:
Not really the best way to solve this problem. Way too hard coded.
This should definitily be changed in the next version.

Reply:
Method naming conventions are violated in this class